tests/python/test_dataset_conversion.py

```diff
@@ -0,0 +1,255 @@
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+
+from smartredis import Dataset
+import numpy as np
+from smartredis.util import DatasetConverter 
+
+import xarray as xr
+
+
+# Create data 
+
+lon = np.linspace(0,360,10)
+lat = np.linspace(-90,90,5)
+lon_coord = (
+'x',
+lon,
+{ 'x_coord_units':'degrees E', 'x_coord_longname':'Longitude'},
+None
+) 
+lat_coord = (
+'y',
+lat,
+{ 'y_coord_units':'degrees N', 'y_coord_longname':'Latitude'},
+None
+)
+data_attributes = {
+'units':'m/s',
+'longname':'velocity',
+'convention':'CF1.5'
+}
+
+data1d = np.random.random([10])
+
+data2d = np.random.random([10,5])
+
+ds_1d = xr.DataArray(
+data=data1d,
+dims='x',
+coords = (lon_coord,), 
+attrs=data_attributes
+)
+
+
+ds_2d = xr.DataArray(
+data=data2d,
+dims=['x','y'],
+coords = (lon_coord,lat_coord),
+attrs=data_attributes
+)
+
+
+# Create Datasets 
+
+ds1 = Dataset("ds-1d")
+
+ds2 = Dataset("ds-2d")
+
+
+  
+def test_add_metadata_for_xarray():
+    
+  
+# Calling add_metadata_for_xarray method on the two dataset exmaples and the list of field names
+
+
+# User adding 1D xarray data to SmartRedis dataset
+
+    ###
+     # 1d data
+    ds1.add_tensor("1ddata",data1d) 
+    
+    # coord 
+    ds1.add_tensor("x",lon)
+    ds1.add_meta_string("x_coord_units","degrees E")    # this might not work actually 
+    ds1.add_meta_string("x_coord_longname","Longitude")
+
+    #data attr 
+    ds1.add_meta_string("units","m/s")
+    ds1.add_meta_string("longname","velocity")
+    ds1.add_meta_string("convention","CF1.5")
+
+    # dims   
+    ds1.add_meta_string("dim_data_x","x")
+
+
+
+
+# Calling the first method for the 1d array 
+    
+    DatasetConverter.add_metadata_for_xarray(
+        ds1, 
+        data_names=["1ddata"],
+        dim_names=["dim_data_x"],
+        coord_names=["x"], # this has to be the same as 
+        attr_names=["units","longname","convention"]
+    )
+    
+    # coordinate 1
+    DatasetConverter.add_metadata_for_xarray(
+    ds1,
+    data_names=["x"], 
+    dim_names=["dim_data_x"],
+    attr_names=["x_coord_units","x_coord_longname"]    # cant have the same name for retrieval 
+    )
+
+    
+    
+
+# Assert that created data and data put into the dataset is the same 
+
+    assert(ds1.get_tensor("1ddata") == data1d).all()
+
+    # # #attr stuff 
```

Comment:
Make these comments more descriptive since to help figure out what might be going on if these tests fail